WriteVHDLSyntax.py

```diff
@@ -634,8 +631,54 @@ def writeTopLevelMemoryType(mtypeB, memList, memInfo, extraports, delay = 0, spl
+            seed = mem.split("_")[1][:-1]
+            iTC = mem.split("_")[1][-1]
+            for PCGroup in MPARdict[seed]:
+                if iTC == PCGroup[0]:
+                    numInputs = len(PCGroup)
+                    merge_parameterlist += "        RAM_WIDTH => "+str(ramwidth)+",\n"
+                    merge_parameterlist += "        NUM_PAGES => "+str(numpages)+",\n"
+                    merge_parameterlist += "        NUM_INPUTS => "+str(numInputs)+",\n"
+                    merge_parameterlist += "        NUM_EXTRA_BITS => 2,\n"
+                    merge_portlist += "        bx_in => TP_bx_out,\n"
+                    merge_portlist += "        rst => '0',\n"
+                    merge_portlist += "        clk => clk,\n"
+                    merge_portlist += "        enb_arr => open,\n"
+                    merge_portlist += "        bx_out => open,\n"
+                    merge_portlist += "        merged_dout => MPAR_"+seed+PCGroup+"_stream_V_dout,\n"
+                    for i in range(4):  merge_portlist += "        din"+str(i)+"=>TPAR_"+seed+PCGroup[i%numInputs]+"_V_dout,\n"
+                    for i in range(4):  merge_portlist += "        nent"+str(i)+"=>TPAR_"+seed+PCGroup[i%numInputs]+"_AV_dout_nent,\n"
+                    for i in range(numInputs):  merge_portlist += "        addr_arr("+str(((i+1)*addrwidth)-1)+" downto "+ str(i*addrwidth)+ ")=>TPAR_"+seed+PCGroup[i%numInputs]+"_V_readaddr,\n"
+                    mem_str += "    MERGE_STREAM_TPAR"+seed+PCGroup+" : entity work.tf_merge_streamer\n"
+                    mem_str += "      generic map (\n"+merge_parameterlist.rstrip(",\n")+"\n      )\n"
+                    mem_str += "      port map (\n"+merge_portlist.rstrip(",\n")+"\n      );\n\n"
+
+        elif "AS" in mem and "n1" in mem and split == 1:
+            addrwidth = 10
+            ramwidth = memInfo.bitwidth
+            numpages = 8
+            numInputs = 1
+            merge_parameterlist += "        RAM_WIDTH => "+str(ramwidth)+",\n"
+            merge_parameterlist += "        NUM_PAGES => "+str(numpages)+",\n"
+            merge_parameterlist += "        NUM_INPUTS => "+str(numInputs)+",\n"
+            merge_parameterlist += "        NUM_EXTRA_BITS => 0,\n"
+            merge_portlist += "        bx_in => TP_bx_out,\n"
+            merge_portlist += "        rst => '0',\n"
+            merge_portlist += "        clk => clk,\n"
+            merge_portlist += "        enb_arr => open,\n"
+            merge_portlist += "        bx_out => open,\n"
+            merge_portlist += "        merged_dout => "+mem+"_stream_V_dout,\n"
+            for i in range(4):  merge_portlist += "        din"+str(i)+"=>" +mem+"_V_dout,\n"
+            for i in range(4):  merge_portlist += "        nent"+str(i)+"=>" +mem+"_AV_dout_nent,\n"
+            for i in range(numInputs):  merge_portlist += "        addr_arr("+str(((i+1)*addrwidth)-1)+" downto "+ str(i*addrwidth)+ ")=>"+mem+"_V_readaddr,\n"
```

Comment:
@jasonfan393 Do we even need to specify the range of the `addr_arr` port here, since there is only one input?:
```suggestion
            ramwidth = memInfo.bitwidth
            numpages = 8
            numInputs = 1
            merge_parameterlist += "        RAM_WIDTH => "+str(ramwidth)+",\n"
            merge_parameterlist += "        NUM_PAGES => "+str(numpages)+",\n"
            merge_parameterlist += "        NUM_INPUTS => "+str(numInputs)+",\n"
            merge_parameterlist += "        NUM_EXTRA_BITS => 0,\n"
            merge_portlist += "        bx_in => TP_bx_out,\n"
            merge_portlist += "        rst => '0',\n"
            merge_portlist += "        clk => clk,\n"
            merge_portlist += "        enb_arr => open,\n"
            merge_portlist += "        bx_out => open,\n"
            merge_portlist += "        merged_dout => "+mem+"_stream_V_dout,\n"
            for i in range(4):  merge_portlist += "        din"+str(i)+"=>" +mem+"_V_dout,\n"
            for i in range(4):  merge_portlist += "        nent"+str(i)+"=>" +mem+"_AV_dout_nent,\n"
            for i in range(numInputs):  merge_portlist += "        addr_arr=>"+mem+"_V_readaddr,\n"
```